# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerPrivate.cpp
# Commit: ea27cd66fefd
# Full Hash: ea27cd66fefd175c69de110191319544a252e820
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-18 09:51:31
# Regressor Bug: 1351231
# File Overlap Count: 1
# Description:
#   Bug 1351231 - Support conversation between InternalRequest and IPCInternalRequest. r=dom-worker-reviewers,jesup
#   
#   Depends on D138812
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138813
# ==============================================================================

diff -r 8c21c9468949 -r ea27cd66fefd dom/serviceworkers/ServiceWorkerPrivate.cpp
--- a/dom/serviceworkers/ServiceWorkerPrivate.cpp	Tue Jan 17 15:13:44 2023 +0000
+++ b/dom/serviceworkers/ServiceWorkerPrivate.cpp	Tue Jan 17 15:13:44 2023 +0000
@@ -324,6 +324,7 @@
 
   nsAutoString referrer;
   ReferrerPolicy referrerPolicy = ReferrerPolicy::_empty;
+  ReferrerPolicy environmentReferrerPolicy = ReferrerPolicy::_empty;
 
   nsCOMPtr<nsIReferrerInfo> referrerInfo = httpChannel->GetReferrerInfo();
   if (referrerInfo) {
@@ -393,14 +394,18 @@
                                                 &isThirdPartyChannel));
   }
 
+  nsILoadInfo::CrossOriginEmbedderPolicy embedderPolicy =
+      loadInfo->GetLoadingEmbedderPolicy();
+
   // Note: all the arguments are copied rather than moved, which would be more
   // efficient, because there's no move-friendly constructor generated.
   return IPCInternalRequest(
       method, {spec}, ipcHeadersGuard, ipcHeaders, Nothing(), -1,
       alternativeDataType, contentPolicyType, referrer, referrerPolicy,
-      requestMode, requestCredentials, cacheMode, requestRedirect, integrity,
-      fragment, principalInfo, interceptionPrincipalInfo, contentPolicyType,
-      redirectChain, isThirdPartyChannel);
+      environmentReferrerPolicy, requestMode, requestCredentials, cacheMode,
+      requestRedirect, integrity, fragment, principalInfo,
+      interceptionPrincipalInfo, contentPolicyType, redirectChain,
+      isThirdPartyChannel, embedderPolicy);
 }
 
 nsresult MaybeStoreStreamForBackgroundThread(nsIInterceptedChannel* aChannel,
